File: atividade_5/primos.c
#include <stdio.h>

int main() {
    int num, i, isPrime = 1; 

    printf("Digite um numero: ");
    scanf("%d", &num);

   
    if (num == 0 || num == 1) {
        isPrime = 0;
        }
    
    else if (num != 2 && num % 2 == 0) {
        isPrime = 0; 
    } 
    else {
       
        for (i = 3; i * i <= num; i += 2) {
            if (num % i == 0) {
                isPrime = 0; 
                break;
            }
        }
    }

    if (isPrime) {
        printf("%d e um numero primo.\n", num);
    } else {
        printf("%d nao e um numero primo.\n", num);
    }

    return 0;
}

File: atividade_4/primos.py
def is_prime(num):
    if num <= 1:
        return False
    elif num == 2:
        return True
    elif num % 2 == 0:
        return False
    else:
        for i in range(3, int(num ** 0.5) + 1, 2):
            if num % i == 0:
                return False
        return True

num = int(input("Digite um numero: "))

if is_prime(num):
    print(f"{num} e um numero primo.")
else:
    print(f"{num} nao e um numero primo.")
